src/targets/gpu/kernels/include/migraphx/kernels/types.hpp

```diff
@@ -25,11 +25,16 @@
 #define MIGRAPHX_GUARD_AMDMIGRAPHX_KERNELS_TYPES_HPP
 
 #include <migraphx/kernels/hip.hpp>
+#include <initializer_list>
 
 namespace migraphx {
 
-using index_int = std::uint32_t;
-using diff_int  = std::int32_t;
+using index_int = uint32_t;
+using diff_int  = int32_t;
+using int8_t    = signed char;
+using uint8_t   = unsigned char;
+using int16_t   = signed short int;
+using uint16_t  = unsigned short int;
```

Comment:
Does `__hip_uint16_t` work?

Reply:
It doesn't just yet. Compiler team has fix merged in and I tested on a docker provided by other team, it works with those fixes, but with released rocm versions, it doesn't work yet.

Reply:
Added required typedefs.